src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalRunListener.java

```diff
@@ -34,6 +40,28 @@ public GlobalRunListener() {
         }
     }
 
+    private void addWorkflowListener(final Run run, final TaskListener listener) {
+        ListenableFuture<FlowExecution> promise = ((WorkflowRun) run).getExecutionPromise();
+        promise.addListener(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    FlowExecution ex = ((WorkflowRun) run).getExecutionPromise().get();
+                    ex.addListener(new GlobalWorkflowListener(run));
+                    /*
+                    * Preferably use catch (InterruptedException | ExecutionException e),
+                    * but requires -source 1.7 flag.
+                    */
+                } catch (InterruptedException e) {
+                    e.printStackTrace(listener.getLogger());
+                    listener.error("Not able to get Workflow listener for this job");
+                } catch (ExecutionException e) {
```

Comment:
This `ExecutionException` could be replaced by generic one - `Exception`. In this case you will be able to remove `InterruptedException` as well.